### PR TITLE
Path decorators: allow empty docstrings to support embedded Python

### PR DESCRIPTION
**Pull request by ameily**
I am developing an application that runs on the Windows embedded Python package. It appears that the embedded Python standard library does not include docstrings which means an unhandled exception is raised when importing trio because of a couple `assert X.__doc__ is not None` statements. This PR makes it so `__doc__` is allowed to be `None`.

To test, I am using [Python Windows Embedded 64-bit 3.11.8](https://www.python.org/downloads/release/python-3118/) package. I wrote a small script to show that `__doc__ is None` on every Path function:

```python
import inspect
from pathlib import Path

for name in dir(Path):
    if name.startswith("_"):
        continue

    attr = getattr(Path, name)
    if inspect.isfunction(attr) or inspect.ismethod(attr):
        if attr.__doc__ is None:
            print(f"Path.{name}.__doc__ is None")
```

When I run this through embedded Python I see every function in `Path` has no docstring.

### CODE REVIEW

**Comment by A5rocks** [comment]
IMO we shouldn't set the docstring if there isn't one, rather than acting as though the docstring is blank.

**Comment by TeamSpen210** [comment]
That would make sense yes. This might be tricky to test for though - Trio can't work in `-OO` mode, because `pycparser` (imported from `cffi`) needs docstrings to work. But it would be fine if only the standard library is stripped.

**Review by A5rocks — CHANGES_REQUESTED** [review]
Sorry, I should have marked my previous comment properly. Here it is just as reference:
> IMO we shouldn't set the docstring if there isn't one, rather than acting as though the docstring is blank.

File: src/trio/_path.py [hunk 1]
```diff
@@ -38,11 +38,10 @@ async def wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
             return await run_sync(partial(fn, *args, **kwargs))
 
         update_wrapper(wrapper, wrapped)
-        assert wrapped.__doc__ is not None
         wrapper.__doc__ = (
             f"Like :meth:`~{wrapped.__module__}.{wrapped.__qualname__}`, but async.\n"
             f"\n"
-            f"{cleandoc(wrapped.__doc__)}\n"
+            f"{cleandoc(wrapped.__doc__ or '')}\n"
```
[the remark this answers is not in the document]
Comment (ameily) [reply src/trio/_path.py]: Sorry about that

**Review by A5rocks — APPROVED** [review]

**Comment by jakkdl** [comment]
Could add tests which monkeypatch `Path` and then runs the decorator on them, which would appease codecov.

**Comment by A5rocks** [comment]
I added a test case cause it can be annoying to make them. cc @jakkdl does it look good?

**Comment by A5rocks** [comment]
@ameily I'm super flakey on these (so you can ignore this if you want) but you don't have a changelog attached! Might be nice to have that.

File: src/trio/_tests/test_path.py [hunk 2]
```diff
@@ -252,3 +252,21 @@ async def test_classmethods() -> None:
...
+        trio._path._wrap_method,
+        trio._path._wrap_method_path,
+        trio._path._wrap_method_path_iterable,
+    ],
+)
+def test_wrapping_without_docstrings(
+    wrapper: Callable[[Callable[[], None]], Callable[[], None]]
+) -> None:
+    @wrapper
+    def func_without_docstring() -> None: ...
```
Comment (jakkdl) [inline src/trio/_tests/test_path.py:270]:
```suggestion
    def func_without_docstring() -> None: ...  # pragma: no cover
```

**Review by jakkdl — CHANGES_REQUESTED** [review]

**Comment by jakkdl** [comment]
> I added a test case cause it can be annoying to make them. cc @jakkdl does it look good?
> 
> (the only iffy bit is passing a sync function to a function that assumes it is async -- but it's not like we run it and we can update that if we ever require passed in functions to satisfy `inspect.iscoroutinefunction`)

looks great! (other than needing a `# pragma: no cover` :P)

**Review by jakkdl — APPROVED** [review]

File: newsfragments/2987.issue.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Fix crash when importing trio in python windows embedded, and other installs that removes docstrings.
```
Comment (A5rocks) [inline newsfragments/2987.issue.rst:1]:
```suggestion
Fix crash when importing trio in embedded Python on Windows, and other installs that remove docstrings.
```

Grammar nitpick

**Review by A5rocks — APPROVED** [review]

File: newsfragments/2987.issue.rst (hunk 3 above)
Comment (CoolCat467) [reply newsfragments/2987.issue.rst]: I would have seconded this

Comment (jakkdl) [reply newsfragments/2987.issue.rst]: (I merged the suggestion in b519645)

**Pull request merged**